cmd/kubelet/app/server.go

```diff
@@ -597,6 +622,94 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) error {
```

Comment:
mutating one clientConfig like this is going to cause mismatches between clients built in different parts of the kubelet init process (there's even a mismatch in this file in the other place we invoke `CreateAPIServerClientConfig(s)`)